docs/source/advanced.rst

```diff
@@ -0,0 +1,354 @@
+.. role:: python(code)
+   :language: python
+
+Advanced
+========
+
+.. _init-model-factory:
+
+:samp:`init_model_factory`
+--------------------------
+
+The :samp:`init_model_factory` interface is less user friendly but perhaps of
+interest to advanced users. It accepts the specification in dictionary format
+(so it has fewer dependencies than :ref:`init-yaml` and :ref:`init-json`) and
+does not construct a declarative base. It accepts the following parameters:
+
+* :samp:`base`: The SQLAlchemy declarative base as a keyword only argument. It
+  is used to as the base class for all SQLAlchemy models.
+* :samp:`spec`: The OpenAPI specification as a dictionary as a keyword only
+  argument.
+* :samp:`models_filename`: The name of the file where the SQLAlchemy models
+  will be written as an optional keyword only argument.
+
+.. note:: the :samp:`define_all` parameter has been removed and OpenAlchemy
+  behaves as though it is set to :samp:`True`.
+
+The return value is the :samp:`model_factory` as defined as part of the return
+value of :ref:`init-yaml`.
+
+.. _build-yaml:
+
+:samp:`build_yaml`
+------------------
+
+Used to build a package with the SQLAlchemy models (including type hints) based
+on a YAML OpenAPI specification which has been extended with any relevant
+OpenAlchemy extension properties.
+
+The :samp:`build_yaml` interface requires the :samp:`PyYAML` library to be
+installed. The :samp:`build_yaml` interface accepts the following arguments:
+
+* :samp:`spec_filename`: The name of the OpenAPI specification file. The file
+  must by a YAML file.
```

Comment:
This wasn't introduced by this change, but I did notice it and it would be great to fix it. This should say `... must be a YAML...` and not `... must by a YAML...`

Reply:
Addressed ✅